Automattic/wp-calypso
# Upgrades: New plans upgrade action

This introduces upgrade action to @lamosty #6119

- Attempts to almost remove `/lib/plans-list`
- Simplifies a bunch
- Adds a reliable route
- Fixes a bug that route `/checkout/jetpack-site/business` allowed to purchase business for jetpack
- Fixes a bug that route `/checkout/any-site/beginner` is just failing.
- Removes a buggy possibility to downgrade without refunding
- Adds resilient routes for checkout that handle both PlanSlugs and Plan Path ids.

### For business site:

![zrzut ekranu 2016-06-28 o 16 26 04](https://cloud.githubusercontent.com/assets/3775068/16419280/6843ae4c-3d4d-11e6-86ce-01344ac1c75b.png)

## Testing

- Run Calypso with `ENABLE_FEATURES=manage/plan-features make run`
- Go to http://calypso.localhost:3000/plans/features/$site
- Click upgrade button under any plan
- Verify that you go to checkout

Make sure following routes work properly

- /checkout/site.wpcom/premium
- /checkout/site.wpcom/personal
- /checkout/site.wpcom/business
- /checkout/site.jetpack/premium
- /checkout/site.jetpack/professional

- make sure that any other route does **not** work.
- make sure all the existing checkout flows still work
- make sure checkouts in nux work
- make sure buing themes work

CC @gwwar @lamosty @rralian 

Test live: https://calypso.live/?branch=update/plans-actions-refactor
